falcon/asgi/app.py

```diff
@@ -460,8 +462,19 @@ async def __call__(self, scope, receive, send):  # noqa: C901
                     'the result to Response.sse, e.g.: resp.sse = some_asyncgen_function()'
                 )
 
+            # NOTE(kgriffs): This must be done in a separate task because
+            #   recieve() can block for some time (until the connection is
+            #   actually closed).
+            async def watch_disconnect():
```

Comment:
While testing the conductor I discovered some subtle bugs related to handling the 'http.disconnect' event, and so I went ahead and fixed them as part of this PR.